src/main/java/io/cdap/plugin/snowflake/common/BaseSnowflakeConfig.java

```diff
@@ -299,7 +307,7 @@ protected void validateConnection(FailureCollector collector) {
         .withConfigProperty(PROPERTY_USERNAME);
 
       // TODO: for oauth2
-      if (keyPairEnabled) {
+      if (Boolean.TRUE.equals(keyPairEnabled)) {
```

Comment:
Why is this change required?

Reply:
keyPairEnabled is a primitive data type with Boolean, and can return NULL, which can lead to NPE.